### ISSUE TITLE
Chainbase preloading "state" database file gets to 200% progress

### ISSUE THREAD

**Issue by matthewdarwin** [EOSIO/eos#8204]
Actual: Progress gets to 200%.
Expected: Progress only gets to 100%

v1.8.5
chain-state-db-size-mb = 24576
database-map-mode = heap
plugin = eosio::state_history_plugin,
on EOS Mainnet

If any other configuration information is required, let me know.

```
Nov  8 16:07:37 eos-sh8 nodeos[108]: CHAINBASE: Preloading "state" database file, this could take a moment...
Nov  8 16:07:38 eos-sh8 nodeos[108]:               1% complete...
Nov  8 16:07:39 eos-sh8 nodeos[108]:               2% complete...
Nov  8 16:07:40 eos-sh8 nodeos[108]:               3% complete...
Nov  8 16:07:41 eos-sh8 nodeos[108]:               4% complete...
Nov  8 16:07:42 eos-sh8 nodeos[108]:               5% complete...
Nov  8 16:07:43 eos-sh8 nodeos[108]:               6% complete...
Nov  8 16:07:44 eos-sh8 nodeos[108]:               7% complete...
...
Nov  8 16:13:06 eos-sh8 nodeos[108]:               196% complete...
Nov  8 16:13:07 eos-sh8 nodeos[108]:               197% complete...
Nov  8 16:13:08 eos-sh8 nodeos[108]:               197% complete...
Nov  8 16:13:09 eos-sh8 nodeos[108]:               198% complete...
Nov  8 16:13:10 eos-sh8 nodeos[108]:               198% complete...
Nov  8 16:13:11 eos-sh8 nodeos[108]:               199% complete...
Nov  8 16:13:12 eos-sh8 nodeos[108]:               199% complete...
Nov  8 16:13:12 eos-sh8 nodeos[108]:            Complete
Nov  8 16:13:12 eos-sh8 nodeos[108]: CHAINBASE: Preloading "reversible" database file, this could take a moment...
Nov  8 16:13:13 eos-sh8 nodeos[108]:               11% complete...
Nov  8 16:13:14 eos-sh8 nodeos[108]:               24% complete...
Nov  8 16:13:15 eos-sh8 nodeos[108]:               31% complete...
Nov  8 16:13:16 eos-sh8 nodeos[108]:               51% complete...
Nov  8 16:13:17 eos-sh8 nodeos[108]:               55% complete...
Nov  8 16:13:18 eos-sh8 nodeos[108]:               75% complete...
Nov  8 16:13:18 eos-sh8 nodeos[108]:            Complete
```

**Comment by matthewdarwin** [comment]
Also v1.8.6

**Comment by matthewdarwin** [comment]
Ok, I think issue is here that previously I had set `chain-state-db-size-mb = 49152`.  Then I changed config.ini to `chain-state-db-size-mb = 24576`. (reducing by 50%)

Now I have erased the state, and started nodeos with blocks + snapshot.  This fixes the problem and when nodeos starts it only gets to 100%.

Seems that nodeos should notice when the state file is larger than what is configured in config.ini and react accordingly (either use the bigger state file as the size or fail to start).
